core/plugins/jdo/common/src/main/java/org/apache/isis/jdo/persistence/IsisTransactionJdo.java

```diff
@@ -338,13 +338,12 @@ private void doFlush() {
 
             if(!pc_snapshot.isEmpty()) {
                 try {
-                    IsisContext.getPersistenceSession().get().execute(pc_snapshot);
-                    for (PersistenceCommand persistenceCommand : pc_snapshot) {
-                        if (persistenceCommand instanceof DestroyObjectCommand) {
-                            final ObjectAdapter adapter = persistenceCommand.onAdapter();
-                            adapter.setVersion(null);
-                        }
-                    }
+                    
+                    PersistenceSession.current(IsisPersistenceSessionJdo.class)
+                    .getFirst()
+                    .get()
+                    .execute(pc_snapshot);
+                    
```

Comment:
could this be injected in instead of looking up from (what I presume is a) ThreadLocal?

Reply:
Good catch! PersistenceSession.current(...) will be deprecated in the future.

Getting rid of ThreadLocals in general is an open task, yet not started. Also note: This class is not managed by Spring yet.